### PR TITLE
Port test/test_datasets.py to use pytest

### PR DESCRIPTION
**Pull request by sallysyw**
Following the discussion in [#3945](https://github.com/pytorch/vision/issues/3945), we should also port `test/test_dataset` to use `pytest`.

In this PR, we replace all the `self.assertXYZ` with python `assert`.

TODO: Replace the `self.subTest()` in these files and finally remove the `unittest.TestCase` inheritance.

### CODE REVIEW

File: test/test_datasets.py [hunk 1]
```diff
@@ -167,23 +168,19 @@ def test_combined_targets(self):
...
                 actual = type(combined_target)
                 expected = type(individual_target)
-                self.assertIs(
-                    actual,
-                    expected,
-                    f"Type of the combined target does not match the type of the corresponding individual target: "
-                    f"{actual} is not {expected}",
-                )
+                assert (
+                    actual == expected
```
Comment (NicolasHug) [inline test/test_datasets.py:181]: according to the original test, I think this should be

```suggestion
                    actual is expected
```

File: test/test_datasets.py [hunk 2]
```diff
@@ -573,33 +570,29 @@ def test_combined_targets(self):
...
```
Comment (NicolasHug) [inline test/test_datasets.py:583]: here as well, this should be an identity check instead of an equality check

**Review by NicolasHug — APPROVED** [review]
Thanks @sallysyw , contrags on your first torchvision PR!
This looks great, there are just 2 assertions that need a minor change. I'll approve anyway, feel free to self-merge once the changes are addressed

**Comment by NicolasHug** [comment]
Indeed the next step will be to fully remove the use of `unittest.TestCase` and thus remove the `subTest` parts, which I think is fine.

It'd be nice to change the `test_all_configs` decorator into a pytest parametrization. @pmeier do you foresee any major complication in doing that?

File: test/test_datasets.py (hunk 1 above)
Comment (sallysyw) [reply test/test_datasets.py]: Thanks for pointing it out!

**Comment by sallysyw** [comment]
Some CI tests are failing but they don't seem related - is it safe to merge? @NicolasHug

**Comment by pmeier** [comment]
> It'd be nice to change the test_all_configs decorator into a pytest parametrization. @pmeier do you foresee any major complication in doing that?

IIRC, you can't use `pytest.mark.parametrize` on the test methods of a `unittest.TestCase`. So we would also need to change that class. But then, we are currently using the `TestCase.setUpClass` method

https://github.com/pytorch/vision/blob/b29ed34f46ba328b46f3aadc5ac15f2a9417a3b0/test/datasets_utils.py#L387-L392

which we would have to emulate another way.

**Comment by NicolasHug** [comment]
> IIRC, you can't use pytest.mark.parametrize on the test methods of a unittest.TestCase

That's correct, and getting rid of `unittest` altogether is our end goal here :). Regarding the setup, we'll probably just have to rename it into a `setup_class` method instead: https://docs.pytest.org/en/6.2.x/xunit_setup.html#class-level-setup-teardown

**Pull request merged**